Test: Using Orchestration Grounding module (#228)
SAP/ai-sdk-java: docs/guides/ORCHESTRATION_CHAT_COMPLETION.md

In this example, the input will be masked before the call to the LLM and will remain masked in the output.

### Grounding

Use the grounding module to provide additional context to the AI model. 

```java
    var message =
        Message.user(
            "{{?groundingInput}} Use the following information as additional context: {{?groundingOutput}}");
    var prompt =
        new OrchestrationPrompt(Map.of("groundingInput", "What does Joule do?"), message);

    var filterInner =
        DocumentGroundingFilter.create().id("someID").dataRepositoryType(DataRepositoryType.VECTOR);
    var groundingConfigConfig =
        GroundingModuleConfigConfig.create()
            .inputParams(List.of("groundingInput"))
            .outputParam("groundingOutput");
    groundingConfigConfig.setFilters(List.of(filterInner));
    
    var groundingConfig =
        GroundingModuleConfig.create()
            .type(GroundingModuleConfig.TypeEnum.DOCUMENT_GROUNDING_SERVICE)
            .config(groundingConfigConfig);
    var configWithGrounding = config.withGroundingConfig(groundingConfig);

    var result =  
            new OrchestrationClient().chatCompletion(prompt, configWithGrounding);
```

> In this example, the AI model is provided with additional context in the form of grounding information. Note, that it is necessary to provide the grounding information in the user message.

Reviewer: (minor) 

```suggestion
In this example, the AI model is provided with additional context in the form of grounding information. Note, that it is necessary to provide the grounding input via one or more input variables.
```

Technically, one could also have the grounding input be part of the system message instead of a user message. However, this is rather uncommon in practice. These docs only need to make clear that using variables is required.
